### PR TITLE
Add path-based negotiation

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: src/EventListener/DefaultLocaleListener.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentNegotiationBundle\EventListener;
+
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+final class DefaultLocaleListener
```
Comment (thewilkybarkid) [inline src/EventListener/DefaultLocaleListener.php:9]: Discovered that the default locale isn't available when handling router errors, opened https://github.com/symfony/symfony/pull/29483.

File: README.md [hunk 2]
```diff
@@ -16,6 +16,51 @@ composer require libero/content-negotation-bundle
 
 If you're not using [Symfony Flex](https://symfony.com/doc/current/setup/flex.html), you'll need to enable the bundle in your application.
 
+### Path-based negotiation
```
Comment (giorgiosironi) [inline README.md:19]: documentation-driven development

File: README.md [hunk 3]
```diff
@@ -16,6 +16,51 @@ composer require libero/content-negotation-bundle
...
+
+You can add negotiation to paths through configuration.
+
+For example, to add an requirement of XML or JSON and English or French to an exact path, and plain text and German elsewhere:
```
Comment (giorgiosironi) [inline README.md]: typo in `an`?

File: tests/Functional/PathFormatNegotiationTest.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        $this->assertSame(200, $response->getStatusCode());
+        $this->assertSame($expected, $response->headers->get('X-Negotiated-Format'));
+        $this->assertSame($expected ?? 'default', $request->getRequestFormat('default'));
+    }
+
+    public function validNegotiateProvider() : iterable
+    {
+        yield ['/foo', '*/*', 'xml'];
+        yield ['/foo/bar', '*/*', 'html'];
```
Comment (giorgiosironi) [inline tests/Functional/PathFormatNegotiationTest.php]: the random paths do not make clear what the data provider scenario is doing without opening separate files that describe them, they could be made readable

File: tests/Functional/PathLocaleNegotiationTest.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        }
+
+        $response = $kernel->handle($request);
+
+        $this->assertSame(200, $response->getStatusCode());
+        $this->assertSame('und', $response->headers->get('X-Negotiated-Locale'));
+        $this->assertSame('und', $request->getLocale());
+    }
+
+    public function anyLocaleProvider() : iterable
```
Comment (giorgiosironi) [inline tests/Functional/PathLocaleNegotiationTest.php:34]: these values look more specific to `Accept-Language` than generic POSIX locales such as `en_US`

**Review by giorgiosironi — APPROVED** [review]

**Pull request merged**